curation/domain_services.py

```diff
@@ -50,26 +46,20 @@ def download_interpro(self):
             raise Exception(e)
         logger.info('InterPro entry list download complete.')
 
-    def get_domain_id(self, name: str) -> str:
+    def get_domain_id(self, name: str) -> Tuple[str, List[str]]:
         """Given functional domain name, return Interpro ID.
         :param str name: name to fetch ID for (case insensitive)
-        :return: domain ID, formatted as CURIE
-        :raises: LookupError if domain ID cannot be retrieved
+        :return: Tuple containing domain ID (as CURIE) if found, empty string otherwise,
+            and a List of warnings (empty if successful)
         """
         domain_id = self.domains.get(name.lower())
         if not domain_id:
-            raise LookupError(f'Functional domain ID lookup failed for {name}')
+            warn = f'Could not retrieve ID for domain {name}'
+            logger.info(warn)
+            return ('', [warn])
```

Comment:
Just out of curiosity, is there a reason why you sometimes return None and sometimes an empty object?

Reply:
Not really, mostly just making easy fixes to make typechecking happy -- we'll end up breaking one way or the other depending on what our typescript model generator thing is capable of handling